### ISSUE TITLE
Proposal for redressing data list usage within viewers

### ISSUE THREAD

**Issue by nmearl** [spacetelescope/jdaviz#105]
Based on conversations out-of-band between myself, @eteq, @dtn5ah, and @rosteen, we've come to some conclusions concerning a few issues about the current usability of the data list panel and the management of data from e.g. within a viewer vs outside a viewer.

The main issues boil down to:

- Access to the data list is non-obvious, and user feedback suggests it is too buried in the UI for easy usage.
- There's currently no UI feedback about data being added to the internal data collection aside from viewing the data list, which access to is non-intuitive.
- Managing data within viewers is currently done via the data list panel (e.g. the ability to add and remove data is tied to the toggle state of the element in the data list). This should instead be done using layers, which is what is actually created when a data set is added to a viewer. It should be noted that data is global, and is not tied to any specific viewer(s), but layers are local to the viewer instance. Therefore, users should be using layers to manage their viewer data, not the checkboxes in a data list.

The major proposed solution is that a new data list that exists outside the viewers should be added that can be used to drag data sets onto a viewer to populate the viewer with the data. This would be easily accessible and obvious, and can perhaps reflect the current number of loaded data sets. This draggable data list does not reflect the state of the data of any particular viewer, and in essence just behaves like the typical QtGlue data list (for interested parties, this is very similar to the original proposed solution at the start of the project, see relevant confluence pages concerning why this approach was not taken originally). This is predominately a UI addition and does not influence the data list access from individual viewers approach currently supported.

A second usability point is the ability to add data to viewers from within viewers. Akin to what is currently supported, however, subtly different in that data management is now done appropriately using layers within a viewer. In this case, data can still be _added_ to a viewer via e.g. a new dropdown within the layer options tab, but the once added, users must remove the layer using the layer tools already provided if they wish to remove the data set from the viewer. This would then support the ability to create multiple layers from the same data set, something that is not currently supported via the data list checkbox approach.

To recap:
1. users can use the global access menu to drag-and-drop data sets onto viewers to create layers within the viewer, and
2. an additional widget within the layer options panel in a specific viewer would allow data to be added to the viewer in the same fashion.
3. Finally, management of data within a viewer would be done via their viewer-specific layer proxy.

**Comment by havok2063** [comment]
I had a discussion with @eteq at the last ASB/DATB touchpoint meeting about my experience (or confusion) with data loading / containers.  He suggested I add my comments here.  My setup was the following:  a single `spectrum-viewer` and a single `image-viewer`, loading two separate data files, one for each viewer.  I had assumed that the `load_data` method meant "load data into a viewer" but it really means "loads data into a generic data container". 

Currently all data loaded is shown in all viewer data tabs, and loading data not meant for a specific viewer breaks the app.  I think the data tab in each viewer should contain only the data relevant to that viewer.  I like the idea of having a separate "data container" dropdown on the main toolbar, containing all data loaded into the app.   If the viewer data tab is restructured to be relevant-data-only, then you could more easily separate data handling and management between the viewer data and layers tabs.
